### PR TITLE
Pytest conversion capsule

### PR DESCRIPTION
**Pull request by vsedmik**
Test results:
```
(venv) [vsedmik@localhost robottelo]$ pytest tests/foreman/api/test_smartproxy.py tests/foreman/cli/test_capsule.py
=================================== test session starts ====================================
platform linux -- Python 3.7.7, pytest-6.1.2, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/vsedmik/PycharmProjects/robottelo, configfile: pyproject.toml
plugins: mock-3.3.1, forked-1.3.0, services-2.2.1, ibutsu-1.13, xdist-2.1.0
collecting ... 2020-12-21 20:11:18 - conftest - DEBUG - Collected 45 test cases
collected 45 items                                                                                                                                                                                                                       

tests/foreman/api/test_smartproxy.py .................                               [ 37%]
tests/foreman/cli/test_capsule.py ...................sssssssss                       [100%]

===================================== warnings summary =====================================
...
================== 36 passed, 9 skipped, 54 warnings in 627.20s (0:10:27) ==================
```

### CODE REVIEW

File: tests/foreman/api/test_smartproxy.py [hunk 1]
```diff
@@ -221,17 +238,16 @@ def test_positive_import_puppet_classes(self):
...
                 "Successfully updated environment and puppetclasses from "
-                "the on-disk puppet installation",
-            )
+                "the on-disk puppet installation"
+            ) in result['message']
 
 
 @pytest.mark.run_in_one_thread
-class SmartProxyMissingAttrTestCase(APITestCase):
+class TestSmartProxyMissingAttr:
```
Comment (JacobCallahan) [inline tests/foreman/api/test_smartproxy.py:250]: does this need to be a separate class? If not, then this entire file should become class-less.

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_smartproxy.py (hunk 1 above)
Comment (vsedmik) [reply tests/foreman/api/test_smartproxy.py]: I can't see any functional reason, apart from keeping the tests logically separated.

File: tests/foreman/api/test_smartproxy.py [hunk 2]
```diff
@@ -22,282 +22,282 @@
...
-        proxy = entities.SmartProxy(**kwargs).create()
-        # Add proxy id to cleanup list
-        self.addCleanup(capsule_cleanup, proxy.id)
-        return proxy
 
-    @skip_if_not_set('fake_capsules')
-    @pytest.mark.tier1
-    def test_negative_create_with_url(self):
-        """Proxy creation with random URL
+@pytest.fixture(scope='class')
```
Comment (JacobCallahan) [inline tests/foreman/api/test_smartproxy.py:35]: what would the correct scope be, now that it isn't used in a class?

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
Thanks for making this class-less!!

File: tests/foreman/api/test_smartproxy.py (hunk 2 above)
Comment (vsedmik) [reply tests/foreman/api/test_smartproxy.py]: Thanks, I missed that one! Module scope fits.

**Review by mshriver — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**